packages/entity-server/src/models/types.ts

```diff
@@ -5,7 +5,13 @@
 
 import { DatabaseModel, DatabaseType } from '@tupaia/database';
 
+export type DbConditional<T> = { [field in keyof T]?: T[field] | T[field][] };
+
+export type Joined<T, U extends string> = {
+  [field in keyof T as field extends string ? `${U}.${field}` : never]: T[field];
+};
```

Comment:
This is really cool, it basically remaps the keys of one type to be prefixed with the given string, eg:
```
Joined<EntityFields, 'ancestor'> =>
{
  'ancestor.id': string;
  'ancestor.code': string;
  'ancestor.name': string;
  ...
}
```